RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -90,6 +93,62 @@ namespace {
     double timeChi2;
   };
 
+  class TrackSegments {
+  public:
+    TrackSegments() {
+      nSegment_ = 0;
+      segmentPathOvc_.clear();
+      segmentMom2_.clear();
+    }
+
+    inline size_t addSegment(double tPath, double tMom2) {
+      segmentPathOvc_.emplace_back(tPath * c_inv);
+      segmentMom2_.emplace_back(tMom2);
+      nSegment_++;
+
+      LogTrace("TrackExtenderWithMTD") << "addSegment # " << nSegment_ << " s = " << tPath
+                                       << " p = " << std::sqrt(tMom2);
+
+      return nSegment_;
+    }
+
+    inline const double computeTof(double mass_inv2) {
+      double tof(0.);
+      for (size_t iSeg = 0; iSeg < nSegment_; iSeg++) {
+        double gammasq = 1. + segmentMom2_[iSeg] * mass_inv2;
+        double beta = std::sqrt(1. - 1. / gammasq);
+        tof += segmentPathOvc_[iSeg] / beta;
+
+        LogTrace("TrackExtenderWithMTD") << " TOF Segment # " << iSeg + 1 << " p = " << std::sqrt(segmentMom2_[iSeg])
+                                         << " tof = " << tof;
+      }
+
+      return tof;
+    }
+
+    inline size_t getSize() const { return nSegment_; }
+
+    inline size_t removeFirstSegment() {
+      if (nSegment_ > 0) {
+        segmentPathOvc_.erase(segmentPathOvc_.begin());
+        segmentMom2_.erase(segmentMom2_.begin());
+        nSegment_--;
+      }
+      return nSegment_;
+    }
+
+    inline std::pair<double, double> getSegment(size_t iSegment) const {
+      if (iSegment >= nSegment_) {
+        throw cms::Exception("TrackExtenderWithMTD") << "Requesting non existing track segment #" << iSegment;
+      }
+      return std::make_pair(segmentPathOvc_[iSegment], segmentMom2_[iSegment]);
+    }
+
+    size_t nSegment_;
```

Comment:
```suggestion
    size_t nSegment_ = 0;
```
is the 64-bit value really needed here? (`unsigned int` could work just as well)

Reply:
I agree, not really needed given the typical size of the underlying arrays